### DISCUSSION TITLE
how to start the web gui of fabric?

### DISCUSSION THREAD

**Discussion by OriLiMu** [danielmiessler/Fabric#854] [Q&A] [answered]
Result of `fabric --help`. No gui option. I watched this video
[New Fabric GUI! AI Augmentation of Crowdsourced #Cybersecurity Job Market Analysis #fabric #ai - YouTube](https://www.youtube.com/watch?v=aEdS8ocmOZs&t=4s). But it says that `fabric --gui` will start the gui mode. But I don't have the gui option
```sh
fabric --help                                                                                                                               ─╯
Usage:
  fabric [OPTIONS]

Application Options:
  -p, --pattern=                    Choose a pattern
  -C, --context=                    Choose a context
      --session=                    Choose a session
  -S, --setup                       Run setup
      --setup-skip-update-patterns  Skip update patterns at setup
  -t, --temperature=                Set temperature (default: 0.7)
  -T, --topp=                       Set top P (default: 0.9)
  -s, --stream                      Stream
  -P, --presencepenalty=            Set presence penalty (default: 0.0)
  -F, --frequencypenalty=           Set frequency penalty (default: 0.0)
  -l, --listpatterns                List all patterns
  -L, --listmodels                  List all available models
  -x, --listcontexts                List all contexts
  -X, --listsessions                List all sessions
  -U, --updatepatterns              Update patterns
  -c, --copy                        Copy to clipboard
  -m, --model=                      Choose model
  -o, --output=                     Output to file
  -n, --latest=                     Number of latest patterns to list (default: 0)
  -d, --changeDefaultModel          Change default pattern

Help Options:
  -h, --help                        Show this help message

```

**Comment by xssdoctor** [comment, marked as answer]
Yeah it’s coming back. I actually made it already in go with a web frontend but it kind of sucks at the moment. It’s a work in progress.

**Reply by LaBalaDeOro** [reply]
Great news! A few questions:
* Did you use a GUI framework? If so, which one?
* What is ETA for release?
* Do you need beta testers?

**Reply by LaBalaDeOro** [reply]
@xssdoctor, I found your Electron GUI in the [trying_something_awesome](https://github.com/danielmiessler/fabric/tree/trying_something_awesome/installer/client/gui) branch.
Surprisingly, after I installed and configured the latest Go version, the gui recognized the current patterns and I able to successfully run a pattern.
Is the GUI you're working on also Electron based?

**Reply by noamsiegel** [reply]
I made my own [here](https://github.com/noamsiegel/fabric-app)

**Comment by xssdoctor** [comment]
Hey! so actually i already wrote a go api with a web frontend but its not working well yet. Its coming I promise. It will expose a bunch of awesome endpoints and then have a vanilla javascript or htmx frontend on top of the api, so you can use the web frontend or do whatever you want with the api

**Comment by noamsiegel** [comment]
I'm also creating a front-end for Fabric. You can see the progress [here](https://github.com/noamsiegel/fabric-app).

**Comment by RepairYourTech** [comment]
whats the stayus on a working gui

**Reply by noamsiegel** [reply]
@mayphilc Releasing mine today/tmmr. See fabric-app on my GitHub repos.
